### PR TITLE
exercise 17

### PR DESCRIPTION
**Pull request by hauviptt97**

### CODE REVIEW

File: src/main/java/Exercise17.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class Exercise17 {
+
+    private int sumOfDigits(int number) {
```
Comment (voquanghoa) [inline src/main/java/Exercise17.java]:
```suggestion
    private int countOfDigits(int number) {
```

File: src/main/java/Exercise17.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        int count = 0;
```
Comment (voquanghoa) [inline src/main/java/Exercise17.java]: init count = 1  then return count

File: src/main/java/Exercise17.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    private boolean isValidNumber(int number) {
+        int sum = 0, a = number;
+        int index = sumOfDigits(number);
+        do {
+            int temp = number % 10;
+            sum += Math.pow(temp, index);
+            number /= 10;
+        } while (number > 0);
+        if (sum == a) {
```
Comment (voquanghoa) [inline src/main/java/Exercise17.java]:
```java
return sum == a;

File: src/main/java/Exercise17.java (hunk 1 above)
Comment (hauviptt97) [reply src/main/java/Exercise17.java]: done

File: src/main/java/Exercise17.java (hunk 2 above)
Comment (hauviptt97) [reply src/main/java/Exercise17.java]: done

File: src/main/java/Exercise17.java (hunk 3 above)
Comment (hauviptt97) [reply src/main/java/Exercise17.java]: done

File: src/main/java/Exercise17.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        int index = countOfDigits(number);
+        do {
+            int temp = number % 10;
+            sum += Math.pow(temp, index);
+            number /= 10;
+        } while (number > 0);
+        return sum == a;
+    }
+
+    public List<Integer> displayResult() {
```
Comment (voquanghoa) [inline src/main/java/Exercise17.java]:
```suggestion
    public List<Integer> getValidNumbers() {
```

Comment (hauviptt97) [reply src/main/java/Exercise17.java]: done

**Pull request merged**